### PR TITLE
Kafka

### PR DESCRIPTION
**Pull request by skothari-tibco**
Add KafkaPub Activity and KafkaSub Trigger

### CODE REVIEW

**Comment by fm-tibco** [comment]
Can you rename these to just Kafka.  Also can you remove any non-kafka related files from the PR ( I think there are some go.mods from other activities)

File: activity/kafka/README.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+```
+
+## Configuration
+
+
+### Input:
+
+| Name       | Type   | Description
+|:---        | :---   | :---   
+| BrokerUrls | string |   |
```
Comment (fm-tibco) [inline activity/kafka/README.md]: Seems like several of these can be settings

File: activity/kafka/README.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+```bash
+flogo install github.com/project-flogo/contrib/activity/kafka
+```
+
+## Configuration
+
+
+### Input:
+
+| Name       | Type   | Description
```
Comment (fm-tibco) [inline activity/kafka/README.md]: Can you add descriptions to these

File: trigger/kafka/README.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+### Flogo CLI
+```bash
+flogo install github.com/project-flogo/contrib/trigger/kafka
+```
+
+## Configuration
+
+### Setting :
+
+| Name       | Type   | Description
```
Comment (fm-tibco) [inline trigger/kafka/README.md]: Add description

File: activity/kafka/README.md (hunk 1 above)
Comment (skothari-tibco) [reply activity/kafka/README.md]: Even I thought so, but in the old-repo it was a part of input. Should I change it to settings similar to what's in the trigger?

Comment (fm-tibco) [reply activity/kafka/README.md]: Think of things this way for the most part, Settings are things that are typically done once and  don't change between executions.  The old repo activities only had input as a way to configure, that is why it was under input.  Please update this to use settings.

File: activity/kafka/README.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+|:---        | :---   | :---
```
Comment (fm-tibco) [inline activity/kafka/README.md]: Can you update the doc to reflect what is a Setting

File: activity/kafka/descriptor.json [hunk 5]
```diff
@@ -0,0 +1,48 @@
+{
+    "name": "tibco-kafkapub",
+    "version": "0.0.1",
+    "title": "Publish Kafka message",
+    "type": "flogo:activity",
+    "description": "Publish a message to a kafka topic",
+    "author": "Wendell Nichols <wnichols@tibco.com>",
+    "settings":[
```
Comment (fm-tibco) [inline activity/kafka/descriptor.json]: Can you add descriptions from the README to these

File: trigger/kafka/descriptor.json [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    "output": [
+      {
+        "name": "message",
+        "type": "string"
+      }
+    ],
+    "handler": {
+      "settings": [
+        {
+          "name": "Topic",
```
Comment (fm-tibco) [inline trigger/kafka/descriptor.json]: Topic should be lower case

File: trigger/kafka/descriptor.json [hunk 7]
```diff
@@ -0,0 +1,53 @@
+{
+    "name": "flogo-kafkasub",
+    "type": "flogo:trigger",
+    "version": "0.0.1",
+    "title": "Receive Kafka Messages",
+    "author": "Wendell Nichols <wnichols@tibco.com>",
+    "description": "Simple Kafka Trigger",
+    "settings":[
+      {
+        "name": "BrokerUrl",
```
Comment (fm-tibco) [inline trigger/kafka/descriptor.json]: should be brokerUrl , add descriptions

File: activity/kafka/README.md [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+| BrokerUrls | string | The Kafka cluster to connect to  |
+| User       | string | If connectiong to a SASL enabled port, the userid to use for authentication  |
+| Password   | string | If connectiong to a SASL enabled port, the password to use for authentication  |
+| TrustStore | string |  If connectiong to a TLS secured port, the directory containing the certificates representing the trust chain for the connection. This is usually just the CACert used to sign the server's certificate |
+
+### Input:
+
+| Name       | Type   | Description
+|:---        | :---   | :---  
+| Topic      | string | The Kafka topic on which to place the message  |
```
Comment (fm-tibco) [inline activity/kafka/README.md]: just have json keys

**Pull request merged**